src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupServiceSpec.scala

```diff
@@ -0,0 +1,123 @@
+  private val defaultRoles = Set(defaultOwnerRole, ResourceRole(ManagedGroupService.MemberRoleName, Set.empty))
+  private val managedGroupResourceType = ResourceType(ManagedGroupService.ManagedGroupTypeName, resourceActionPatterns, defaultRoles, ownerRoleName)
+  private val resourceTypes = Map(managedGroupResourceType.name -> managedGroupResourceType)
+  private val resourceService = new ResourceService(resourceTypes, policyDAO, dirDAO, NoExtensions, "example.com")
+
+  val managedGroupService = new ManagedGroupService(resourceService, resourceTypes)
+
+  private val dummyUserInfo = UserInfo(OAuth2BearerToken("token"), WorkbenchUserId("userid"), WorkbenchEmail("user@company.com"), 0)
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    runAndWait(schemaDao.init())
+  }
+
+  def makeResourceType(): ResourceTypeName = runAndWait(resourceService.createResourceType(managedGroupResourceType))
+
+  def makeGroup(groupId: String = resourceId.value): Resource = {
+    makeResourceType()
+    runAndWait(managedGroupService.createManagedGroup(ResourceId(groupId), dummyUserInfo))
+  }
+
+  before {
+    runAndWait(schemaDao.clearDatabase())
+    runAndWait(schemaDao.createOrgUnits())
+  }
+
+  "ManagedGroupService create" should "create a managed group with admin and member policies" in {
+    makeGroup()
+    val policies = runAndWait(policyDAO.listAccessPolicies(expectedResource))
+    policies.map(_.id.accessPolicyName.value) shouldEqual Set("admin", "member")
+  }
+
+  it should "create a workbenchGroup with the same name as the Managed Group" in {
+    makeGroup()
+    val samGroup: Option[BasicWorkbenchGroup] = runAndWait(dirDAO.loadGroup(WorkbenchGroupName(resourceId.value)))
+    samGroup.value.id.value shouldEqual resourceId.value
+  }
+
+  it should "create a workbenchGroup with 2 member WorkbenchSubjects" in {
+    makeGroup()
+    val samGroup: Option[BasicWorkbenchGroup] = runAndWait(dirDAO.loadGroup(WorkbenchGroupName(resourceId.value)))
+    samGroup.value.members shouldEqual Set(adminPolicy, memberPolicy)
+  }
+
+  it should "fail when the group name is too long" in {
+    val groupName = "a" * 64
+    intercept[WorkbenchExceptionWithErrorReport] {
+      makeGroup(groupName)
+    }
+    runAndWait(managedGroupService.loadManagedGroup(resourceId)).isDefined shouldEqual false
+  }
+
+  it should "fail when the group name has invalid characters" in {
+    val groupName = "Make It Rain!!! $$$$$"
+    intercept[WorkbenchExceptionWithErrorReport] {
+      makeGroup(groupName)
+    }
+    runAndWait(managedGroupService.loadManagedGroup(resourceId)).isDefined shouldEqual false
+  }
+
+  "ManagedGroupService get" should "return the Managed Group resource" in {
+    makeGroup()
+    runAndWait(managedGroupService.loadManagedGroup(resourceId)).isDefined shouldEqual true
+  }
+
+  // NOTE: All since we don't have a way to look up policies directly without going through a Resource, this test
```

Comment:
policies are sub contexts of resources so this is not possible give the current ldap structure